smolyay/basis.py

```diff
@@ -72,6 +80,31 @@ def in_domain(self, x):
             numpy.greater_equal(x, self.domain[0]), numpy.less_equal(x, self.domain[1])
         )
 
+    def scale_to_domain(self, points, old_domain):
+        """Scale points from a domain to BasisFunction domain
+
+        Parameters
+        ----------
+        points: numeric or array-like
+            points to be shifted to new domain.
+
+        old_domain: ndarray of shape (2,)
+            upper and lower bounds of points.
+
+        Returns
+        -------
+        numeric or array-like
+            points shifted to BasisFunction domain"""
+        new_points = numpy.asarray(
+            self.domain[0]
+            + (self.domain[1] - self.domain[0])
+            * ((points - old_domain[0]) / (old_domain[1] - old_domain[0]))
```

Comment:
Won't this fail if `points` is a list or tuple? I think you need to cast `points` to a NumPy array, then you won't need `asarray` here.

Reply:
Fixed